src/TypeSystem/Inferrer/InferredTypes.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * PackageFactory.ComponentEngine - Universal View Components for PHP
+ *   Copyright (C) 2022 Contributors of PackageFactory.ComponentEngine
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types=1);
+
+namespace PackageFactory\ComponentEngine\TypeSystem\Inferrer;
+
+use PackageFactory\ComponentEngine\TypeSystem\TypeInterface;
+
+class InferredTypes
+{
+    /**
+     * Map of identifierName to the corresponding inferred type
+     * @var array<string,TypeInterface>
+     */
+    private readonly array $types;
+
+    private function __construct(
+        TypeInterface ...$types
+    ) {
+        assert(self::isAssociativeArray($types), '$types must be an associative array');
```

Comment:
```suggestion
```

It's fair to assume that it is. The class itself is in control of this (via static factory methods).

I know, it's not perfect, since anyone who changes this class must be aware of this implicit assertion. Yet, on the other hand: What can really go wrong here? If `$types` has numeric keys, they will be ignored - nothing breaks.

Reply:
Yes was just trying out other things :D It would be cool if you could annotate and tell phpstan, that the the varadic/spread should be with string keys. So it is statically enforced.